magicctapipe/conftest.py

```diff
@@ -0,0 +1,660 @@
+
+
+@pytest.fixture(scope="session")
+def dl0_p_path(base_path):
+    return base_path / "DL0p"
+
+
+@pytest.fixture(scope="session")
+def dl0_M1_path(base_path):
+    return base_path / "DL0M1"
+
+
+@pytest.fixture(scope="session")
+def dl0_M2_path(base_path):
+    return base_path / "DL0M2"
+
+
+@pytest.fixture(scope="session")
+def dl1_lst_path(base_path):
+    return base_path / "DL1LST"
+
+
+@pytest.fixture(scope="session")
+def conf_path(base_path):
+    p = base_path / "config"
+    return p
+
+
+"""
+Custom data
+"""
+
+
+@pytest.fixture(scope="session")
+def dl2_test(temp_DL2_test):
+    """
+    Toy DL2
+    """
+    path = temp_DL2_test / "dl2_test.h5"
+    data = Table()
+    data["obs_id"] = [1, 1, 2, 2, 2, 3, 3]
+    data["event_id"] = [7, 7, 8, 8, 8, 7, 7]
+    data["tel_id"] = [1, 2, 1, 2, 3, 1, 3]
+    data["combo_type"] = [1, 1, 3, 3, 3, 2, 2]
+    data["multiplicity"] = [2, 2, 3, 3, 3, 2, 2]
+    data["timestamp"] = [1, 1, 4, 4, 4, 10, 10]
+    data["pointing_alt"] = [0.6, 0.6, 0.7, 0.7, 0.7, 0.5, 0.5]
+    data["pointing_az"] = [1.5, 1.5, 1.5, 1.5, 1.5, 1.5, 1.5]
+    data["reco_energy"] = [1, 1, 1, 1, 1, 1, 1]
+    data["gammaness"] = [0.5, 0.5, 1, 0.3, 0.5, 1, 1]
+    data["reco_alt"] = [40, 40, 41, 41, 41, 42, 42]
+    data["reco_az"] = [85, 85, 85, 85, 85, 85, 85]
+    write_table_hdf5(
+        data, str(path), "/events/parameters", overwrite=True, serialize_meta=False
+    )
+    return path
+
+
+@pytest.fixture(scope="session")
+def pd_test():
+    """
+    Toy dataframe
+    """
+    df = pd.DataFrame(np.array([[1, 2], [3, 4], [5, 6]]), columns=["a", "b"])
+    return df
+
+
+"""
+Remote paths (to download test files)
+"""
+
+
+@pytest.fixture(scope="session")
+def base_url():
+    return Path("cp02:/fefs/aswg/workspace/elisa.visentin/git_test")
+
+
+@pytest.fixture(scope="session")
+def dl0_gamma_url(base_url):
+    return base_url / "DL0gamma"
+
+
+@pytest.fixture(scope="session")
+def dl0_p_url(base_url):
+    return base_url / "DL0p"
+
+
+@pytest.fixture(scope="session")
+def dl0_M_url(base_url):
+    return base_url / "MAGIC"
+
+
+@pytest.fixture(scope="session")
+def dl1_LST_url(base_url):
+    # BETTER SOLUTION? CHANGED?
+    return base_url / "LST"
+
+
+@pytest.fixture(scope="session")
+def conf_url(base_url):
+    q = base_url / "config.yaml"
+    return q
+
+
+"""
+Downloads: useful functions
+"""
+
+
+@pytest.fixture(scope="session")
+def env_prefix():
+    # ENVIRONMENT VARIABLES TO BE CREATED
+    return "MAGIC_CTA_DATA_"
+
+
+def scp_file(path, url, env_prefix):
+    """
+    Download of test files through rsync
+    """
+    pwd = os.environ[env_prefix + "PASSWORD"]
+    if not (path / url.name).exists():
+        print("DOWNLOADING...")
+        cmd = f'''/bin/bash -c "rsync {str(url)} {str(path)}"'''
+        if "rm " in cmd:
+            print("files cannot be removed")
+            exit()
+        usr = os.environ[env_prefix + "USER"]
+        childP = pexpect.spawn(cmd, timeout=5000)
```

Comment:
at least in my machine rsync does not create the needed parent directory, so in order to make the files being downloaded I needed to add:
```subprocess.run(["mkdir", "-p", str(path)])```

Reply:
This is another issue due to the SSH connection (IT Container). Test data should be downloaded to a temporary directory created on the fly while running the tests and deleted at the end of the session (as the ones used to store data). Now, the IT Container is really busy and often runs into crashes, so, to avoid to keep on downloading the same data (the administrators won't be happy) I coded more 'persistent' downloads. 
But I will fix the directory creation (I didn't noticed this, thank you really much :)  )